src/pages/keyword-generator/KeywordGenerator.vue

```diff
@@ -1,161 +1,492 @@
 <script setup>
-import { ref, computed } from 'vue'
+import { ref, computed, watch } from 'vue'
+import { 
+  generateKeywords as generateKeywordsUtil, 
+  getNGramOptions, 
+  getNGramClass 
+} from '@/utils/keywordGenerator.js'
+import { MaInput, MaTagInput } from "@mobileaction/action-kit"
+
 
-// Reactive data
 const inputText = ref('')
-const generatedKeywords = ref({
+const selectedNGrams = ref([1, 2, 3]) 
+const generatedKeywords = ref({})
+
+const keywordTags = ref({
   '1-gram': [],
   '2-gram': [],
-  '3-gram': []
+  '3-gram': [],
+  '4-gram': [],
+  '5-gram': [],
+  '6-gram': [],
+  '7-gram': [],
+  '8-gram': [],
+  '9-gram': [],
+  '10-gram': []
 })
 
-// Text cleaning function
-const cleanText = (text) => {
-  return text
-    .toLowerCase()
-    .replace(/[^\w\s]/g, ' ') // Remove special characters except spaces
-    .replace(/\s+/g, ' ') // Replace multiple spaces with single space
-    .trim()
-}
-
-// Generate n-gram keywords
-const generateNGrams = (words, n) => {
-  const ngrams = []
-  for (let i = 0; i <= words.length - n; i++) {
-    const ngram = words.slice(i, i + n).join(' ')
-    ngrams.push(ngram)
-  }
-  return ngrams
+// Generates keywords from input text
+const generateKeywords = () => {
+  const keywords = generateKeywordsUtil(
+    inputText.value,
+    selectedNGrams.value
+  )
+  
+  generatedKeywords.value = keywords
 }
 
-// Generate keywords from input text
-const generateKeywords = () => {
-  if (!inputText.value.trim()) {
-    generatedKeywords.value = {
-      '1-gram': [],
-      '2-gram': [],
-      '3-gram': []
-    }
-    return
-  }
 
-  const cleanedText = cleanText(inputText.value)
-  const words = cleanedText.split(' ').filter(word => word.length > 0)
+watch(generatedKeywords, (newKeywords) => {
+  selectedNGrams.value.forEach(n => {
+    const key = `${n}-gram`
+    keywordTags.value[key] = newKeywords[key] || []
+  })
+}, { deep: true })
```

Comment:
```suggestion
const keywordTags = computed(() => {
  return selectedNGrams.value.reduce((acc, n) => {
    const key = `${n}-gram`
    acc[key] = generatedKeywords.value[key] || []
    return acc
  }, {})
```
This can be improved by implementing keywordTags as computed. and remove the watcher